### PR TITLE
Spam Email (Send multiple email fix)

### PR DESCRIPTION
**Pull request by Lt-Kartik-9PARA-SF**
send multiple email fix

### CODE REVIEW

File: pages/api/v1/auth/actions/reportContent.ts [hunk 1]
```diff
@@ -83,14 +84,27 @@ async function handler(req: NextApiRequest, res: NextApiResponse<IReportContentR
...
+				console.log('keySet comment');
+				_sendCommentReportMail(network,strPostType,post_id,comment_id,totalUsers);
+				await redisSetex('commentReportKey',60,'true');
+			}
 		}
 		if(type == 'reply' && checkReportThreshold(totalUsers) ){
 			_sendReplyReportMail(network,strPostType,post_id,comment_id,reply_id,totalUsers);
 		}
+		//const key =  await redisGet('emailKey');
+		console.log('email key =>',key);
```
[the remark this answers is not in the document]
Comment (Lt-Kartik-9PARA-SF) [reply pages/api/v1/auth/actions/reportContent.ts:107]: resolved

File: pages/api/v1/auth/actions/reportContent.ts [hunk 2]
```diff
@@ -83,14 +84,27 @@ async function handler(req: NextApiRequest, res: NextApiResponse<IReportContentR
 		const totalUsers = data.count || 0;
 
 		if(type == 'post' && checkReportThreshold(totalUsers) ){
-			_sendPostSpamReportMail(network,strPostType,contentId,totalUsers);
+			const postReportKey = await redisGet('postReportKey');
+			if(!postReportKey){
+				console.log('keySet');
```
[the remark this answers is not in the document]
Comment (Lt-Kartik-9PARA-SF) [reply pages/api/v1/auth/actions/reportContent.ts:89]: resolved

File: pages/api/v1/auth/actions/reportContent.ts [hunk 3]
```diff
@@ -83,14 +84,27 @@ async function handler(req: NextApiRequest, res: NextApiResponse<IReportContentR
...
+				_sendPostSpamReportMail(network,strPostType,contentId,totalUsers);
+				await redisSetex('postReportKey',21600,'true');
+			}
 		}
 		if(type == 'comment' && checkReportThreshold(totalUsers) ){
-			_sendCommentReportMail(network,strPostType,post_id,comment_id,totalUsers);
+
+			const commentReportKey = await redisGet('commentReportKey');
+			if(!commentReportKey){
+				console.log('keySet comment');
```
[the remark this answers is not in the document]
Comment (Lt-Kartik-9PARA-SF) [reply pages/api/v1/auth/actions/reportContent.ts:98]: resolved

File: pages/api/v1/auth/actions/reportContent.ts [hunk 4]
```diff
@@ -83,13 +84,26 @@ async function handler(req: NextApiRequest, res: NextApiResponse<IReportContentR
 		const totalUsers = data.count || 0;
 
 		if(type == 'post' && checkReportThreshold(totalUsers) ){
-			_sendPostSpamReportMail(network,strPostType,contentId,totalUsers);
+			const postReportKey = await redisGet(`postReportKey-${post_id}`);
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/reportContent.ts:87]: post_id can be same for networks and even post types

please use `postReportKey-${network}_${strPostType}_${post_id}`

File: pages/api/v1/auth/actions/reportContent.ts [hunk 5]
```diff
@@ -83,13 +84,26 @@ async function handler(req: NextApiRequest, res: NextApiResponse<IReportContentR
...
+			if(!postReportKey){
+				_sendPostSpamReportMail(network,strPostType,contentId,totalUsers);
+				await redisSetex(`postReportKey-${post_id}`,21600,'true');
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/reportContent.ts:90]: please use `postReportKey-${network}_${strPostType}_${post_id}`

File: pages/api/v1/auth/actions/reportContent.ts [hunk 6]
```diff
@@ -83,13 +84,26 @@ async function handler(req: NextApiRequest, res: NextApiResponse<IReportContentR
...
+			}
 		}
 		if(type == 'comment' && checkReportThreshold(totalUsers) ){
-			_sendCommentReportMail(network,strPostType,post_id,comment_id,totalUsers);
+
+			const commentReportKey = await redisGet(`commentReportKey-${comment_id}`);
+			if(!commentReportKey){
+				_sendCommentReportMail(network,strPostType,post_id,comment_id,totalUsers);
+				await redisSetex(`commentReportKey-${comment_id}`,21600,'true');
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/reportContent.ts:98]: please use `commentReportKey-${network}_${strPostType}_${post_id}_${comment_id}`

File: pages/api/v1/auth/actions/reportContent.ts [hunk 7]
```diff
@@ -83,13 +84,26 @@ async function handler(req: NextApiRequest, res: NextApiResponse<IReportContentR
...
+			}
 		}
 		if(type == 'reply' && checkReportThreshold(totalUsers) ){
-			_sendReplyReportMail(network,strPostType,post_id,comment_id,reply_id,totalUsers);
+			const replyReportKey = await redisGet(`replyReportKey-${reply_id}`);
+			if(!replyReportKey){
+				_sendReplyReportMail(network,strPostType,post_id,comment_id,reply_id,totalUsers);
+				await redisSetex(`replyReportKey-${reply_id}`,21600,'true');
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/reportContent.ts:105]: please use `replyReportKey-${network}_${strPostType}_${post_id}_${comment_id}_${reply_id}`

File: pages/api/v1/auth/actions/reportContent.ts (hunk 6 above)
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/reportContent.ts:98]: please avoid using magic numbers and make a const for it

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Please check comments.

File: pages/api/v1/auth/actions/reportContent.ts (hunk 4 above)
Comment (Lt-Kartik-9PARA-SF) [reply pages/api/v1/auth/actions/reportContent.ts:87]: resolved

File: pages/api/v1/auth/actions/reportContent.ts (hunk 6 above)
Comment (Lt-Kartik-9PARA-SF) [reply pages/api/v1/auth/actions/reportContent.ts:98]: resolved

Comment (Lt-Kartik-9PARA-SF) [reply pages/api/v1/auth/actions/reportContent.ts]: resolved

**Review by alphainfinitus — APPROVED** [review]

**Pull request merged**